test/e2e/controller_test.go

```diff
@@ -20,34 +20,80 @@ import (
 	"context"
 	"fmt"
 	"strings"
-	"time"
-
-	"github.com/onsi/ginkgo"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/onsi/ginkgo"
 	clientset "k8s.io/client-go/kubernetes"
 	"k8s.io/kubernetes/test/e2e/framework"
-
-	"sigs.k8s.io/azuredisk-csi-driver/pkg/apis/azuredisk/v1alpha1"
 	azDiskClientSet "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/client/clientset/versioned"
-	"sigs.k8s.io/azuredisk-csi-driver/test/e2e/testsuites"
 )
 
 const (
 	partitionKey = "azdrivernodes.disk.csi.azure.com/partition"
 )
 
 var _ = ginkgo.Describe("Controller", func() {
+	ginkgo.Context("[single-az]", func() {
+		defineTests(false)
+	})
+
+	//TODO add support for scheduler extender
+	ginkgo.Context("[multi-az]", func() {
+		// skip multi zone test for now
+		// TODO
+		//defineTests(true)
+	})
+})
+
+// TODO: migrate to integration test suite
+var defineTests = func(isMultiZone bool) {
 	f := framework.NewDefaultFramework("azuredisk")
 
 	var (
-		cs           clientset.Interface
+		cs clientset.Interface
+		// ns           *v1.Namespace
 		azDiskClient *azDiskClientSet.Clientset
-		err          error
+		// testDriver   driver.PVTestDriver
+		err error
 	)
 
+	// volumeNameSeed := "test-volume-"
+
+	// var setUpPod = func(size, maxShares int) (*testsuites.TestPod, []func()) {
+	// 	pvcSize := strconv.Itoa(size)
+	// 	t := dynamicProvisioningTestSuite{}
+	// 	pod := testsuites.PodDetails{
+	// 		Cmd: convertToPowershellorCmdCommandIfNecessary("echo 'hello world' > /mnt/test-1/data && grep 'hello world' /mnt/test-1/data"),
+	// 		Volumes: t.normalizeVolumes([]testsuites.VolumeDetails{
+	// 			{
+	// 				ClaimSize: pvcSize + "Gi",
+	// 				MountOptions: []string{
+	// 					"barrier=1",
+	// 					"acl",
+	// 				},
+	// 				VolumeMount: testsuites.VolumeMountDetails{
+	// 					NameGenerate:      volumeNameSeed,
+	// 					MountPathGenerate: "/mnt/test-",
+	// 				},
+	// 			},
+	// 		}, isMultiZone),
+	// 		IsWindows: isWindowsCluster,
+	// 	}
+	// 	storageClassParameters := map[string]string{"skuName": "Premium_LRS", "maxShares": strconv.Itoa(maxShares)}
+	// 	tpod, cleanups := pod.SetupWithDynamicVolumes(cs, ns, testDriver, storageClassParameters, schedulerName)
+	// 	tpod.Create()
+	// 	tpod.WaitForSuccess()
+
+	// 	return tpod, cleanups
+	// }
+
+	// testDriver = driver.InitAzureDiskDriver()
```

Comment:
Remove commented out code.

Reply:
I will remove the comments in `controller_test.go` as soon as the migration to integration test suite is completed :)